File: rubix/apps/provision-app-flutter/lib/core/theme/statuses.dart
import 'package:flutter/material.dart';

/// Device/connection STATUS — the repurposed "mood" layer. A live status tints
/// the app accent on top of the chosen theme, so the whole UI reflects the
/// agent connection state at a glance. null status = use the theme accent.
/// Ported from the React app's `theme/statuses.ts`.
enum StatusKey { online, pairing, fault, offline }

@immutable
class AppStatus {
  const AppStatus({required this.key, required this.label, required this.accent});
  final StatusKey key;
  final String label;
  final Color accent;
}

const statuses = <StatusKey, AppStatus>{
  // DNA kit semantic dots: connected=green, warning=amber, offline=red,
  // idle=grey.
  StatusKey.online:
      AppStatus(key: StatusKey.online, label: 'Connected', accent: Color(0xFF5DCAA5)),
  StatusKey.pairing:
      AppStatus(key: StatusKey.pairing, label: 'Provisioning', accent: Color(0xFFC9A24A)),
  StatusKey.fault:
      AppStatus(key: StatusKey.fault, label: 'Offline', accent: Color(0xFFE24B4A)),
  StatusKey.offline:
      AppStatus(key: StatusKey.offline, label: 'Idle', accent: Color(0xFF7E8888)),
};

File: rubix/apps/provision-app/src/theme/statuses.ts
// Device/connection STATUS — the repurposed "mood" layer (§11.3). A live status
// tints the app accent on top of the chosen theme, so the whole UI reflects the
// agent connection state at a glance. null status = use the theme accent.
export type StatusKey = 'online' | 'pairing' | 'fault' | 'offline'

export interface AppStatus {
  key: StatusKey
  label: string
  accent: string
}

export const STATUSES: Record<StatusKey, AppStatus> = {
  online: { key: 'online', label: 'Connected', accent: '#36e2c4' },
  pairing: { key: 'pairing', label: 'Provisioning', accent: '#ffc24b' },
  fault: { key: 'fault', label: 'Fault', accent: '#ff5a52' },
  offline: { key: 'offline', label: 'Offline', accent: '#7c8a8a' },
}
